### ISSUE TITLE
STmag and ABmag cannot be initialized from string

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#5015]
When constructing a Unit using FITS header keyword (say, `TUNITn`), it is convenient to pass in the unit name into `Unit` constructor:

```python
>>> from astropy import units as u
>>> u.Unit(u.AA.to_string())
Unit("Angstrom")
```

But I cannot figure out how to do this with `STmag` or `ABmag`:

```python
>>> u.Unit(u.STmag.to_string())
.../astropy/units/core.pyc in __call__(self, s, represents, format, namespace, doc, parse_strict)
   1798                            .format(s, format_clause, six.text_type(e)))
   1799                     if parse_strict == 'raise':
-> 1800                         raise ValueError(msg)
   1801                     elif parse_strict == 'warn':
   1802                         warnings.warn(msg, UnitsWarning)

ValueError: 'mag(ST)' did not parse as unit: At col 4, ST is not a valid unit.
Did you mean St, YSt, ZSt, ySt or zSt?

>>> u.Unit(u.ABmag.to_string())
.../astropy/units/core.pyc in __call__(self, s, represents, format, namespace, doc, parse_strict)
   1798                            .format(s, format_clause, six.text_type(e)))
   1799                     if parse_strict == 'raise':
-> 1800                         raise ValueError(msg)
   1801                     elif parse_strict == 'warn':
   1802                         warnings.warn(msg, UnitsWarning)

ValueError: 'mag(AB)' did not parse as unit: At col 4, AB is not a valid unit.
Did you mean aB, aWb, ab, daB or dab?
```

@mhvk , any idea?

**Comment by pllim** [comment]
It also seems weird that these units are not recognized as a proper instance of Unit:

```python
>>> isinstance(u.STmag, u.UnitBase)
False
>>> isinstance(u.STmag, u.LogUnit)
True
```

**Comment by mhvk** [comment]
The main problem is that `ST` and `AB` are not properly defined as units. Personally, I'd be happy to add them, but there was some resistance in a long distant past, since `u.ST` seemed a little odd. I do think this should be made to work, so I labeled this as a bug.

Note that even with this fixed, there is a separate question on how to store these in fits headers; see #3822.

p.s. On the classes: when I implemented the function units, it turned out to be very hard to get them to work if they also had to be `UnitBase` instances. In principle, I still hope to get them be part of an abstract unit class at least (see #3650).

**Comment by mhvk** [comment]
See #5030 for a possible solution.  Am not quite sure anymore whether this should be counted as a bug, since it does not make the units available by default....

**Comment by pllim** [comment]
@mhvk , I guess it is not a bug since only one person to date (me) is requesting for it? :smile: 

Thanks for clarifying about the parent class. Until #3650 is addressed, my current workaround is:
```python
if isinstance (unitclass, (u.UnitBase, u.LogUnit)):
    # do something
```

**Comment by mhvk** [comment]
@pllim - what you can do is duck-type, by passing whatever you get through `u.Unit` -- function units are passed through those fine because I don't use `isinstance` there but rather [this](https://github.com/astropy/astropy/blob/master/astropy/units/core.py#L1738) (but don't copy that!).

**Comment by pllim** [comment]
Round-tripping fixed in #5030.
